app/actions/websocket.ts

```diff
@@ -0,0 +1,1157 @@
+        case WebsocketEvents.REACTION_ADDED:
+            return dispatch(handleReactionAddedEvent(msg));
+        case WebsocketEvents.REACTION_REMOVED:
+            return dispatch(handleReactionRemovedEvent(msg));
+        case WebsocketEvents.EMOJI_ADDED:
+            return dispatch(handleAddEmoji(msg));
+        case WebsocketEvents.LICENSE_CHANGED:
+            return dispatch(handleLicenseChangedEvent(msg));
+        case WebsocketEvents.CONFIG_CHANGED:
+            return dispatch(handleConfigChangedEvent(msg));
+        case WebsocketEvents.OPEN_DIALOG:
+            return dispatch(handleOpenDialogEvent(msg));
+        }
+
+        return {data: true};
+    }
+}
+
+function handleNewPostEvent(msg: WebSocketMessage) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState();
+        const currentChannelId = getCurrentChannelId(state);
+        const post = JSON.parse(msg.data.post);
+        const actions: Array<GenericAction> = [];
+
+        const exists = selectPost(state, post.pending_post_id);
+        if (!exists) {
+            if (getCurrentChannelId(state) === post.channel_id) {
+                EventEmitter.emit(WebsocketEvents.INCREASE_POST_VISIBILITY_BY_ONE);
+            }
+
+            const myChannel = getChannel(state, post.channel_id);
+            if (!myChannel) {
+                const channel = await fetchMyChannel(post.channel_id);
+                if (channel.data) {
+                    actions.push({
+                        type: ChannelTypes.RECEIVED_CHANNEL,
+                        data: channel.data,
+                    });
+                }
+            }
+
+            const myChannelMember = selectMyChannelMember(state, post.channel_id);
+            if (!myChannelMember) {
+                console.log('The channel member?')
+                const member = await fetchMyChannelMember(post.channel_id);
+                if (member.data) {
+                    actions.push({
+                        type: ChannelTypes.RECEIVED_MY_CHANNEL_MEMBER,
+                        data: member.data,
+                    });
+                }
+            }
+
+            actions.push(receivedNewPost(post));
+            // If we don't have the thread for this post, fetch it from the server
+            // and include the actions in the batch
+            if (post.root_id) {
+                const rootPost = selectPost(state, post.root_id);
+
+                if (!rootPost) {
+                    const thread: any = await dispatch(getPostThread(post.root_id, true));
+                    if (thread.data?.length) {
+                        actions.push(...thread.data);
+                    }
+                }
+            }
+
+            if (post.channel_id === currentChannelId) {
+                const id = post.channel_id + post.root_id;
+                const {typing} = state.entities;
+
+                if (typing[id]) {
+                    actions.push({
```

Comment:
Maybe this should be a separate dispatch and then queue the other ones to be dispatch after a few milliseconds, that may help with the animation causing the post list jump. @migbot thoughts about that?